Tests/Listener/DeprecationListenerTest.php

```diff
@@ -0,0 +1,129 @@
+        $interactor->expects($this->once())->method('noticeException')->with($this->isInstanceOf(DeprecationException::class));
+
+        $listener = new DeprecationListener($interactor);
+
+        set_error_handler(function () { return false; });
+        try {
+            $listener->register();
+            @trigger_error('This is a deprecation', E_USER_DEPRECATED);
+        } finally {
+            $listener->unregister();
+            restore_error_handler();
+        }
+    }
+
+    public function testDeprecationIsReportedRegardlessErrorReporting()
+    {
+        $interactor = $this->getMockBuilder(NewRelicInteractorInterface::class)->getMock();
+        $interactor->expects($this->once())->method('noticeException');
+
+        $listener = new DeprecationListener($interactor);
+
+        set_error_handler(function () { return false; });
+        $e = error_reporting(0);
+        try {
+            $listener->register();
+            @trigger_error('This is a deprecation', E_USER_DEPRECATED);
+        } finally {
+            $listener->unregister();
+            error_reporting($e);
+            restore_error_handler();
+        }
+    }
+
+    public function testOtherErrorAreIgnored()
+    {
+        $interactor = $this->getMockBuilder(NewRelicInteractorInterface::class)->getMock();
+        $interactor->expects($this->never())->method('noticeException');
+
+        $listener = new DeprecationListener($interactor);
+
+        set_error_handler(function () { return false; });
+        try {
+            $listener->register();
+            @trigger_error('This is a notice', E_USER_NOTICE);
+        } finally {
+            $listener->unregister();
+            restore_error_handler();
+        }
+    }
+
+    public function testInitialHandlerIsCalled()
+    {
+        $interactor = $this->getMockBuilder(NewRelicInteractorInterface::class)->getMock();
+        $interactor->expects($this->once())->method('noticeException');
+
+        $handler = $this->createPartialMock(\stdClass::class, ['__invoke']);
+        $handler->expects($this->once())->method('__invoke');
+
+        $listener = new DeprecationListener($interactor);
+
+        set_error_handler($handler);
+        try {
+            $listener->register();
+            @trigger_error('This is a deprecation', E_USER_DEPRECATED);
+        } finally {
+            $listener->unregister();
+            restore_error_handler();
+        }
+    }
+
+    public function testUnregisterRemovesHandler()
+    {
+        $interactor = $this->getMockBuilder(NewRelicInteractorInterface::class)->getMock();
+        $interactor->expects($this->never())->method('noticeException');
+
+        $listener = new DeprecationListener($interactor);
+
+        set_error_handler(function () { return false; });
+        try {
+            $listener->register();
+            $listener->unregister();
+            @trigger_error('This is a notice', E_USER_NOTICE);
```

Comment:
This should be a deprecation

Reply:
What do you mean?

Reply:
Don't worry, I also comment my code (as a reminder).

The purpose of this test is to assert that when calling `unregister`, the deprecation is not pushed to new relic. But given, this line is triggering a `E_USER_NOTICE` the event won't, in any way, be pushed to newrelic